doc/dev/dataplane_codegen_quick_start.md

```diff
@@ -0,0 +1,198 @@
+# Getting Started - Generate SDK With Dataplane Codegen
+
+## Prerequisites
+
+- [Python 3.7](https://www.python.org/downloads/windows/) or later is required
+
+- [Nodejs 14.x.x](https://nodejs.org/download/release/latest-v14.x/) or later is required
+
+- [Autorest][https://github.com/Azure/autorest/blob/main/docs/install/readme.md]
+
+## Setup your repo
+
+- Fork and clone the azure-sdk-for-python repo.
+
+- Create a branch to work in.
+
+## Project service name, module name, and package name
+
+Three key pieces of information for your project are the `service_name`, `module_name`, and `package_name`.
+
+The `service_name` is the short name for the Azure service. The `service_name` should match across all the SDK language repos
+and is generally name of the directory in the specification folder of the azure-rest-api-specs repo that contains the REST API definition file.
```

Comment:
nit: "generally the name"

Also just spitballing: should we include examples here? "For example, azure-rest-api-specs/specification/resource" corresponds to `resource` being the `service_name`"

Reply:
We should include an example here. I thought I would just use Web Pub Sub since we also use it below.  But I'm not sure it is a good example.
- the API definitions for Web Pub Sub are stored within the `specifications/webpubsub` directory of azure-rest-api-specs.  That suggests that the `service_name` should be "webpubsub".
- but deconstructing the package name a few lines down based on the template shown indicates that the actual `service_name` for Web Pub Sub is "messaging".

Which may be fine to give as an example that _doesn't_ follow the general rule. But it makes me wonder whether there are more exceptions than there are cases that follow the rule.

I think I need to soften the language here, which is disappointing but maybe necessary to reflect reality.